source/include/lostpart.f90

```diff
@@ -32,7 +32,7 @@
   if(llost) then
     ! report losses to user
     call aperture_reportLoss(n,i,ix)
-    call compactArrays
+    if(.not.apflag) call compactArrays
```

Comment:
this same `if` statement should be added also in `collimation.f90`, at `call compactArrays`

Reply:
Hmm, that causes a circular dependency in collimation. Is the apflag variable not internal to aperture? If not, I do have to move it to one of the common modules.

Reply:
ok - no worries. Actually, I've just realised that that check is necessary for losses in the aperture, not for losses in collimators